cypher-shell/src/main/java/org/neo4j/shell/state/BoltStateHandler.java

```diff
@@ -361,17 +367,19 @@ public void changePassword( @Nonnull ConnectionConfig connectionConfig )
             connectionConfig.setPassword(connectionConfig.newPassword());
             connectionConfig.setNewPassword(null);
 
-            // Save a system bookmark to make sure we wait for the password change to propagate on reconnection
-            systemBookmark = session.lastBookmark();
-
             silentDisconnect();
         } catch (Throwable t) {
             try {
                 silentDisconnect();
             } catch (Exception e) {
                 t.addSuppressed(e);
             }
-            throw t;
+            if (t instanceof RuntimeException) {
```

Comment:
I don't understand this part, why not just `throw t`. I don't see that the signature of the method changed?

Reply:
I cannot `throw t` without adding `throws CommandException` to the signature (because `reconnect` has that in its signature) . But we now that Exception won't be thrown, because we supply an empty Command.

Reply:
ok, it would be nicer then to have a `reconnectWithoutPing` method I think.